src/cmap/auth/mongo_credentials.ts

```diff
@@ -134,26 +134,6 @@ export class MongoCredentials {
     this.mechanism = options.mechanism || AuthMechanism.MONGODB_DEFAULT;
     this.mechanismProperties = options.mechanismProperties || {};
 
-    if (this.mechanism.match(/MONGODB-AWS/i)) {
```

Comment:
Does this work because we outsource the env credential reading to the sdk as of NODE-6988 and previously we were overriding the natural order by explicitly reading in the env vars?

Reply:
Yes, reading credentials from the environment in addition to using the SDK has always been a bug (https://jira.mongodb.org/browse/NODE-6987)